pybop/optimisers/base_optimiser.py

```diff
@@ -199,6 +201,76 @@ def _run(self):
         """
         raise NotImplementedError
 
+    def cost_call(self, inputs: Union[Inputs, list], calculate_grad: bool = False):
```

Comment:
This has a lot of duplicate code from the cost class, as such it could indicate that the application of the transformations in better suited there.

Reply:
The optimiser.cost_call function is added to ensure that the error catches occur only when the cost is called by the optimiser. When a user calls the cost for an invalid parameter set, I believe it should return the error.